receiver/awsfirehosereceiver/internal/unmarshaler/cwmetricstream/unmarshaler_test.go

```diff
@@ -45,18 +57,16 @@ func TestUnmarshal(t *testing.T) {
 		"WithSomeInvalidRecords": {
 			filename:           "some_invalid_records",
 			wantResourceCount:  5,
-			wantMetricCount:    35,
+			wantMetricCount:    36,
```

Comment:
Why?  Was there a latent defect or is this papering over a regression?

Reply:
Yeah there was a defect, from the description:

> Due to using the above merging functions, this PR also fixes a bug in the cwmetrics unmarshaller where the unit of a metric was not considered part of its identity, causing two metrics that differed only by unit to be merged.

Reply:
Specifically, the MetadataNoToken metric in some_invalid_records has different units: in most cases it's "None", in one case it's "Count".